accounts-db/src/accounts_db.rs

```diff
@@ -2427,7 +2427,7 @@ pub struct PubkeyHashAccount {
 }
 
 impl AccountsDb {
-    pub const ACCOUNTS_HASH_CACHE_DIR: &str = "accounts_hash_cache";
+    pub const ACCOUNTS_HASH_CACHE_DIR: &'static str = "accounts_hash_cache";
```

Comment:
🤔 I'm confused by this. Thought there was a clippy lint in the opposite direction, i.e. removing these unecessary lifetimes...`redundant_static_lifetimes`.

Wonder why is this one required.

Reply:
Apparently, it is coming from the compiler, not from clippy: https://github.com/rust-lang/rust/issues/115010
Seems like there are some issues with how the lifetimes in associated constants are desugared.  And the compiler team is trying to fix it now.

At the same time, clippy lint has a good example of why it might get verbose, if all lifetimes need to be explicit:

```rust
const FOO: &'static [(&'static str, &'static str, fn(&Bar) -> bool)] =
&[...]
```

https://rust-lang.github.io/rust-clippy/master/index.html#/const_static_lifetime